src/SAML2/Assertion.php

```diff
@@ -906,7 +878,10 @@ public function decryptAttributes(XMLSecurityKey $key, array $blacklist = []): v
             }
 
             if (!array_key_exists($name, $this->attributes)) {
-                $this->attributes[$name] = [];
+                $attr = new Attribute();
+                $attr->setName($name);
```

Comment:
It's a pity that the `Attribute` API uses a `DOMElement` as the parameter to the constructor. With perspective, I think we should've had a static method `fromXML()` in that class, that would return a new `Attribute` object initialized from the XML. Then the constructor would be able to receive the actual properties as arguments. That way all these code could be simplified to:

```php
$attr = new Attribute($name);
```

Most importantly, it would get rid of the inconsistency that we have today, which allows us to create an attribute without a name. This pattern makes it impossible to have objects without proper values for their mandatory properties, making the code more resilient.

Maybe it's a good time to change this behaviour as well?

Reply:
Yes, now is the time to do so!